File: apps/backend/app/api/v1/schemas/range_levels.py
"""Schemas for GET /range-levels — live range buy/sell/stop monitoring."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class RangeLevelRow(BaseModel):
    symbol: str
    buy: float | None      # entry / support (strategy's actual level)
    sell: float | None     # exit / resistance
    stop: float | None
    current_price: float | None
    position_qty: float
    # forming | levels_set | in_range | at_buy | at_sell | below_stop | holding
    status: str
    levels_at: datetime | None  # when the strategy last published these levels


class RangeLevelsResponse(BaseModel):
    strategy_id: int | None
    strategy_name: str | None
    as_of: datetime
    rows: list[RangeLevelRow]

File: apps/frontend/src/api/rangeLevels.ts
import { apiFetch } from "./client";

export interface RangeLevelRow {
  symbol: string;
  buy: number | null;
  sell: number | null;
  stop: number | null;
  current_price: number | null;
  position_qty: number;
  status: string;
  levels_at: string | null;
}

export interface RangeLevelsResponse {
  strategy_id: number | null;
  strategy_name: string | null;
  as_of: string;
  rows: RangeLevelRow[];
}

export const rangeLevelsApi = {
  list(strategyId?: number): Promise<RangeLevelsResponse> {
    const q = strategyId != null ? `?strategy_id=${strategyId}` : "";
    return apiFetch<RangeLevelsResponse>(`/api/v1/range-levels${q}`);
  },
};
